src/onegov/org/views/reservation.py

```diff
@@ -790,6 +790,50 @@ def reject_reservation(self, request, text=None, notify=False):
         }
     )
 
+    # get all recipients which want to receive e-mail for rejected reservations
+    q = ResourceRecipientCollection(request.session).query()
+    q = q.filter(ResourceRecipient.medium == 'email')
+    q = q.order_by(None).order_by(ResourceRecipient.address)
+    q = q.with_entities(ResourceRecipient.address, ResourceRecipient.content)
+    recipients = [
+        r.address
+        for r in q
+        if (
+            self.resource.hex in r.content['resources']
+            and r.content.get('rejected_reservations', {})
+        )
+    ]
```

Comment:
Consider changing this to a generator expression, since you don't use the list for anything.

You could also further improve things by using `send_transactional_email_batch`, which is more efficient than calling `send_transactional_email` multiple times, since each send will generate a file on the disk and a datamanager in the current transaction. Look for other places where we use `send_transactional_email_batch` to see how this works.

Reply:
I think merging this with master somehow brought this back to an older version.